debezium-core/src/main/java/io/debezium/connector/AbstractSourceInfo.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.debezium.connector;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.errors.ConnectException;
+
+/**
+ * Common information provided by all connectors in either source field or offsets
+ * 
+ * @author Jiri Pechanec
+ *
+ */
+public abstract class AbstractSourceInfo {
+    public static final String DEBEZIUM_VERSION_KEY = "version";
+
+    private final String version;
+
+    protected static SchemaBuilder schemaBuilder() {
+        return SchemaBuilder.struct().field(DEBEZIUM_VERSION_KEY, Schema.OPTIONAL_STRING_SCHEMA);
+    }
+
+    protected AbstractSourceInfo() {
+        final String connector = getClass().getName().split("\\.")[3];
```

Comment:
I'm not of encoding an expecation on the naming structure of subclasses and their packaging here. I'll change it a bit so that sub-classes pass the Module up to the constructor here, avoiding all that reflection hazzle. Good find with `Module#getVersion()` btw., wasn't aware of that.